### DISCUSSION TITLE
NER confidence on pre-trained spaCy models

### DISCUSSION THREAD

**Discussion by ivyleavedtoadflax** [explosion/spaCy#9172] [Help: Other Questions] [answered]
Hi Explosion! I'd like to be able to get NER confidence scores on the existing spaCy models. I know that this is possible when training my own models with the new Span Categoriser component, but if I don't have data and just want to use the vanilla models, is there a way to do this? If not, is this ever something that will be possible.

In the meantime I realise one approach I can take is to run predictions using the vanilla models on some data, and use this as my training data to train a model with my own pipeline that includes the Span Categoriser. Does that make sense, is there another way?

**Comment by polm** [comment, marked as answer]
> if I don't have data and just want to use the vanilla models, is there a way to do this?

No, the NER component doesn't expose confidence scores.

> If not, is this ever something that will be possible.

It isn't something we're actively working on.

> In the meantime I realise one approach I can take is to run predictions using the vanilla models on some data, and use this as my training data to train a model with my own pipeline that includes the Span Categoriser. Does that make sense, is there another way?

You can do that. Accuracy will degrade some, but if you have enough data it would usually be usable.

**Reply by polm** [reply]
The overview of the architecture here explains why we don't expose the NER scores, they aren't really "confidence" scores and won't be useful:

https://github.com/explosion/spaCy/issues/831
